### ISSUE TITLE
Trying to install glad for use in Xcode for Mac/osx, not recognizing <glad/glad.h>

### ISSUE THREAD

**Issue by futurepr0n** [Dav1dde/glad#261]
Hey! I was trying to install glad on my Mac so that I can use it with Xcode for some OpenGL tutorials.

I used the unix instructions and installed:
![Screen Shot 2020-04-03 at 4 20 35 PM](https://user-images.githubusercontent.com/478986/78401581-0e796580-75c7-11ea-8e44-38936d922bbe.png)

But it looks like when I reloaded Xcode and tried to use the libraries im still unable to find it.

![Screen Shot 2020-04-03 at 4 19 19 PM](https://user-images.githubusercontent.com/478986/78401504-e1c54e00-75c6-11ea-8d43-5d67fbb1d15a.png)

Is there something I need to do, like add it to search paths in Xcode, and if so, where would I look?

Thanks!

**Comment by futurepr0n** [comment]
I decided to use the site https://glad.dav1d.de, and selected:

OpenGL 3.3 only, All Extensions and the checkbox
Generate a Loader
I copied the glad folder and glad.h to my /usr/local/include and that made me able to find the library in Xcode, my next problem was that on compile I got a bunch of errors.

I had my first error with 
![Screen Shot 2020-04-03 at 5 12 07 PM](https://user-images.githubusercontent.com/478986/78405325-420bbe00-75ce-11ea-8924-fb510ceaf26d.png)

And I then just changed that to typed unsigned int GLhandleARB to match what the else condition was, and it got further, but I would have these other errors which I was wondering if are related to the KHR stuff.

When I retried to build a bundle, and omit the KHR - I still was unable to build,, all sorts of errors related to Khronos platform (Khronos specific ints and the like)


![Screen Shot 2020-04-03 at 5 15 04 PM](https://user-images.githubusercontent.com/478986/78405546-b0e91700-75ce-11ea-8b4a-386f5d8d6de6.png)

Would there be any way to get passed these?

**Comment by teburt** [comment]
Yes, there was a recent issue someone else brought this up: https://github.com/Dav1dde/glad/issues/258#issuecomment-604334663

I'm not sure where your exact issue is, but there are preprocessor directives that should make it so the system opengl headers will not conflict with glad.

**Comment by futurepr0n** [comment]
hmm, I tried adding the compiler flag as mentioned, but it still has issue with all these 


![Screen Shot 2020-04-03 at 6 14 38 PM](https://user-images.githubusercontent.com/478986/78409167-fe698200-75d6-11ea-9328-9319ba001d0b.png)

**Comment by teburt** [comment]
Assuming you don't include OpenGL headers in your file, then you might need this one too, right before you include GLFW: `#define GLFW_INCLUDE_NONE`

**Comment by Dav1dde** [comment]
In your screenshot you include `glfw` before you include `glad` this needs to be the other way round or you define `GLFW_INCLUDE_NONE` as @teburt mentioned. See the bottom of the section the GLFW docs: https://www.glfw.org/docs/3.3/build_guide.html#build_include

**Comment by Dav1dde** [comment]
I assume this worked out for you, feel free to re-open/comment if you have any questions.

**Comment by futurepr0n** [comment]
sorry, was a bad weekend and I got a flu.

Unfortunately this did not fix the issue, I am still having all sorts of Khronos_int errors
![Screen Shot 2020-04-05 at 3 52 20 PM](https://user-images.githubusercontent.com/478986/78508518-6fcf3f00-7755-11ea-8f57-27462e63a239.png)

Do you have recommendations on how to get rid of those?

**Comment by Dav1dde** [comment]
Why are you including glad and glew?

Try compiling the hellowindow2 example

**Comment by futurepr0n** [comment]
thanks @Dav1dde, basically I found a tutorial, which im struggling to find again right now, but it had me download and install both, basically to install from brew GLFW and glew, but then as I was following learnopengl.com and other sites, I came across glad in the tutorial.

Most of my problems were first successfully setting up OS X, and Xcode to build properly, but after getting past that, I went to the OpenGL tutorials and since most have glad.h, this is where I've been struggling to compile successfully since.

I did use the Omit KHR option when bringing these over that was what I tried originally, but let me try again - when going to build your example I got same stuff unfortunately.

![Screen Shot 2020-04-05 at 4 42 48 PM](https://user-images.githubusercontent.com/478986/78509523-7d3bf780-775c-11ea-9def-c8252fb88d7a.png)

Now rebuilding glad, and overwriting the glad.h

I am using GL version 3.3 - because my GLFW is The current version is 3.3.2, so this time, here are all my options.
![Screen Shot 2020-04-05 at 4 44 50 PM](https://user-images.githubusercontent.com/478986/78509561-c5f3b080-775c-11ea-960b-d1d8e37314d3.png)

What do I do with glad.c? Is it necessary? I put it in the same directory as the project main.cpp, but even when building your example, I am still seeing that it for some reason wants those Khronos int

![Screen Shot 2020-04-05 at 4 52 24 PM](https://user-images.githubusercontent.com/478986/78509685-e839fe00-775d-11ea-908e-a73db91c93ee.png)

**Comment by futurepr0n** [comment]
I retried with adding latest KHR and glad and seem to have a new problem but further along 
![Screen Shot 2020-04-05 at 6 50 58 PM](https://user-images.githubusercontent.com/478986/78511909-64d4d880-776e-11ea-85cd-561cd9f4e4c7.png)

**Comment by futurepr0n** [comment]
I managed to compile successfully! But the issue seems to be failing to create. GLFW Window, I will have to search into this more! thank you for your help

![Screen Shot 2020-04-05 at 7 14 11 PM](https://user-images.githubusercontent.com/478986/78512331-a31fc700-7771-11ea-941a-b2ca54e08aac.png)

**Comment by futurepr0n** [comment]
I added this line:

```
#ifdef __APPLE__
        glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE); // uncomment this statement to fix compilation on OS X
    #endif

```
Inside of the int main and it worked!

![Screen Shot 2020-04-05 at 7 44 45 PM](https://user-images.githubusercontent.com/478986/78512862-eaa85200-7775-11ea-82ee-5226a055f07c.png)

**Comment by Dav1dde** [comment]
@futurepr0n awesome, glad you could figure it out. Sorry i wasn't much help, but it seems like we're on opposite sides of the globe
